src/sentry/tagstore/snuba/backend.py

```diff
@@ -333,6 +342,7 @@ def get_tag_keys_for_projects(
             end,
             include_values_seen=False,
             use_cache=use_cache,
+            dataset=Dataset.Discover,
```

Comment:
Because we currently default to the Events dataset in https://github.com/getsentry/sentry/blob/master/src/sentry/utils/snuba.py#L594 this is really the only change.
Added events everywhere else just so that we start being explicit with the dataset being used